test/controllers/sections/items/ItemProducerSizeControllerSpec.scala

```diff
@@ -0,0 +1,212 @@
+          endYear = "2023"
+        )(dataRequest(request), messages(request)).toString
+      }
+
+      "must redirect to ItemsIndexController for a GET there is no GoodsType in UserAnswers" in new Setup() {
+
+        val request = FakeRequest(GET, itemProducerSizeRoute(0))
+
+        val result: Future[Result] = TestController.onPageLoad(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual controllers.sections.items.routes.ItemsIndexController.onPageLoad(testErn, testDraftId).url
+      }
+
+      "must redirect to Journey Recovery for a GET if no existing data is found" in new Setup(None) {
+
+        val request = FakeRequest(GET, itemProducerSizeRoute(0))
+
+        val result: Future[Result] = TestController.onPageLoad(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual controllers.routes.JourneyRecoveryController.onPageLoad().url
+      }
+
+    }
+
+    "for a POST onSubmit" - {
+
+      "must redirect to the next page when valid data is submitted" in new Setup() {
+
+        MockUserAnswersService.set().returns(Future.successful(emptyUserAnswers))
+
+        val request = FakeRequest(POST, itemProducerSizeRoute(0))
+          .withFormUrlEncodedBody(("value", "1"))
+
+        val result: Future[Result] = TestController.onSubmit(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual testOnwardRoute.url
+      }
+
+      "must return a Bad Request and errors when invalid data is submitted and there is a GoodsType set" in new Setup(Some(
+        emptyUserAnswers.set(ItemExciseProductCodePage(0), Wine.code)
+      )) {
+
+        val request = FakeRequest(POST, itemProducerSizeRoute(0))
+          .withFormUrlEncodedBody(("value", ""))
+
+        val boundForm = form.bind(Map("value" -> ""))
+
+        val result: Future[Result] = TestController.onSubmit(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual BAD_REQUEST
+        contentAsString(result) mustEqual view(
+          form = boundForm,
+          onSubmitAction = onSubmitAction(0),
+          goodsType = Wine,
+          startYear = "2022",
+          endYear = "2023"
+        )(dataRequest(request), messages(request)).toString
+      }
+
+      "must redirect to ItemsIndexController when invalid data is submitted and there is a No GoodsType set" in new Setup() {
+
+        val request = FakeRequest(POST, itemProducerSizeRoute(0))
+          .withFormUrlEncodedBody(("value", ""))
+
+        val result: Future[Result] = TestController.onSubmit(testErn, testDraftId, 0, NormalMode)(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual controllers.sections.items.routes.ItemsIndexController.onPageLoad(testErn, testDraftId).url
+      }
+
```

Comment:
Could you add the test for index out of bounds? Should be a copy-paste from other controller tests